cx/spark/server.js

```diff
@@ -0,0 +1,208 @@
+/**
+ * To-Do:
+ * Create a Service Account and download its key file. 
+ * Set the environmental variable GOOGLE_APPLICATION_CREDENTIALS 
+ * to the key file's location.
+ * See https://cloud.google.com/dialogflow/cx/docs and 
+ * https://cloud.google.com/dialogflow/cx/docs/quick/setup for details.
+ */
+
+const express = require('express');
+const request = require('request');
+const app = express();
+
+app.use(express.json());
+
+// Uncomment and insert your values here
+// const sparkAccessToken = "1234567898-ABCdfghTtaD8dfghdfgh-45sdf65467M";
+// const targetUrl = 'https://example.com';
+// const projectId = 'my-project';
+// const locationId = 'global';
+// const agentId = 'my-agent';
+// const languageCode = 'en'
+
+// Imports the Google Cloud Some API library
+const {SessionsClient} = require('@google-cloud/dialogflow-cx');
+/**
+ * Example for regional endpoint:
+ *   const locationId = 'us-central1'
+ *   const client = new SessionsClient({apiEndpoint: 'us-central1-dialogflow.googleapis.com'})
+ */
+const client = new SessionsClient({apiEndpoint: locationId + '-dialogflow.googleapis.com'});
+
+// Upon start a webhook is registered with spark
+// Upon closure the webhook is removed from spark
+
+const listener = app.listen(process.env.PORT, async function() {
+  await init();
+  console.log('Your Spark integration server is listening on port '
+      + listener.address().port);
+});
+
+app.post('/', async function(req, res) {
+  const message = await retrieveMessage(req.body.data.id);
+  const dialogflowResponse = await detectIntentText(message, req.body.data.personId);
+  const sparkMessage = detectIntentToSparkMessage(dialogflowResponse);
+  sendMessage(sparkMessage.text, message.email);
+});
+
+process.on('SIGTERM', () => {
+  listener.close(async ()=>{
+    console.log('Closing http server.');
+    await deleteWebhooks();
+    process.exit(0);
+  });
+});
+
+async function init(){
+  await deleteWebhooks();
+  registerWebhook();
+}
+
+// Converts Spark message to a detectIntent request. 
+function sparkToDetectIntent(message, sessionPath){
+     const request = {
+     session: sessionPath,
+     queryInput: {
+         text: {
+             text: message.text,
+            },
+            languageCode,
+        },
+    };
```

Comment:
nit: wrong indent?

Reply:
The indent seems not fixed? It should be 2-space indent like other functions.

https://google.github.io/styleguide/jsguide.html#formatting-block-indentation
> Each time a new block or block-like construct is opened, the indent increases by two spaces.

Reply:
Just for your information, the follow tools may help on the code style format:
* https://github.com/google/closure-library/wiki/Formatting-.js-with-clang-format
* https://www.npmjs.com/package/eslint-config-google